From c7cafe348f25fd94b46cc1b627426393cce92d57 Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Tue, 21 Jun 2011 12:43:41 +0200
Subject: [PATCH 027/154] h264parse: avc input must either pass-through or be
 split into bytestream chunks

... since parsing code is not meant to receive avc data.

Fixes #653087.
---
 gst/videoparsers/gsth264parse.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index f8b403e..76d3f8f 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -1042,12 +1042,13 @@ gst_h264_parse_set_caps (GstBaseParse * parse, GstCaps * caps)
 
     /* we did parse codec-data and might supplement src caps */
     gst_h264_parse_update_src_caps (h264parse, caps);
-  } else if (format == GST_H264_PARSE_FORMAT_AVC &&
-      h264parse->format == GST_H264_PARSE_FORMAT_BYTE) {
+  } else if (format == GST_H264_PARSE_FORMAT_AVC) {
+    /* if input != output, and input is avc, must split before anything else */
     /* arrange to insert codec-data in-stream if needed.
      * src caps are only arranged for later on */
     h264parse->push_codec = TRUE;
     h264parse->split_packetized = TRUE;
+    h264parse->packetized = TRUE;
   }
 
   return TRUE;
-- 
1.7.5.4

